### PR TITLE
Add PowerShell unit test cases for when service principals are passed into scubagear

### LINKED ISSUE #169: Add PowerShell Unit Test Cases for when Service Principals are passed into ScubaGear
# 💡 Summary #

#167 doesn't including test cases for when Service principals are passed in. This issue for adding test cases for scenarios of when Service Principal related parameters are passed in.

## Motivation and context ##
There currently isn't unit test case coverage for when Service Principals are being utilized in ScubaGear functions. Test cases are needed in this area because it is a blind spot in our unit test coverage. 

## Implementation notes ##
Add test case scenarios for `Invoke-SCuBA.Tests.ps1, Connect-Tenant.Tests.ps1, Connect-DefenderHelper.Tests.ps1` etc. 
Tests can be located under the 'Testing\Unit\PowerShell` folder once #167 is merged in.

## Acceptance criteria ##
Test cases are added for when ` -AppID -CertificateThumbprint -Organization` are passed into ScubaGear functions.

### PR DESCRIPTION
**Pull request by crutchfield**
#  #
  

## 🗣 Description ##

closes #169
Added additional unit test to cover passing/processing of service principal parameters.

## 💭 Motivation and context ##

Unit test cases were added to increase coverage. 

## 🧪 Testing ##

Executed PowerShell unit test, all passed

## ✅ Pre-approval checklist ##

- [X] This PR has an informative and human-readable title.
- [X] PR targets the correct parent branch (e.g., main or release-name) for merge.
- [X] Changes are limited to a single goal - *eschew scope creep!*
- [X] Changes are sized such that they do not touch excessive number of files.
- [X] *All* future TODOs are captured in issues, which are referenced in code comments.
- [X] These code changes follow the ScubaGear [content style guide](https://github.com/cisagov/ScubaGear/blob/main/CONTENTSTYLEGUIDE.md).
- [X] Related issues these changes resolve are linked preferably via [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [X] All relevant type-of-change labels added.
- [X] All relevant project fields are set.
- [X] All relevant repo and/or project documentation updated to reflect these changes.
- [X] Unit tests added/updated to cover PowerShell and Rego changes.
- [X] Functional tests added/updated to cover PowerShell and Rego changes.
- [X] All relevant functional tests passed.
- [X] All automated checks (e.g., linting, static analysis, unit/smoke tests) passed.

## ✅ Pre-merge checklist ##

- [x] PR passed smoke test check.
- [x] Feature branch has been rebased against changes from parent branch, as needed

  Use `Rebase branch` button below or use [this](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request) reference to rebase from the command line.
- [x] Resolved all merge conflicts on branch
- [x] Notified merge coordinator that PR is ready for merge via comment mention

## ✅ Post-merge checklist ##

- [ ] Feature branch deleted after merge to clean up repository.
- [ ] Verified that all checks pass on parent branch (e.g., main or release-name) after merge.

### CODE REVIEW

**Comment by rgbrow1949** [comment]
Your edits and the new test file looks good. I ran the pester tests but got the following error: 

```
Get-ChildItem : Cannot find path 
'C:\Users\rgbrow\ScubaGear\PowerShell\ScubaGear\Testing\Unit\PowerShell\PowerShell\ScubaGear\Sample-Config-Files' 
because it does not exist.
At 
C:\Users\rgbrow\ScubaGear\PowerShell\ScubaGear\Testing\Unit\PowerShell\Support\Copy-ScubaSampleConfigFile.Tests.ps1:5 
char:25
+ ... pleFiles = (Get-ChildItem PowerShell/ScubaGear/Sample-Config-Files).N ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\User    s\rgbrow...le-Config-Files:String) [Get-ChildItem]    , 
ItemNotFoundException
    + FullyQualifiedErrorId : PathNotFound,Microsoft.P    owerShell.Commands.GetChildItemCommand
```

All tests pass, none fail.

Also, this branch is currently out of date with main. 

When the error gets resolved (it may be on my side, I'm not sure) and the branch gets updated, this is ready to be approved.

File: PowerShell/ScubaGear/Testing/Unit/PowerShell/Connection/Connect-Tenant.Tests.ps1 [hunk 1]
```diff
@@ -1,21 +1,22 @@
...
 
 InModuleScope Connection {
     Describe -Tag 'Connection' -Name "Connect-Tenant" -ForEach @(
         @{Endpoint = 'commercial'}
-        @{Endpoint = 'gcc'}
-        @{Endpoint = 'gcchigh'}
-        @{Endpoint = 'dod'}
+        # @{Endpoint = 'gcc'}
+        # @{Endpoint = 'gcchigh'}
+        # @{Endpoint = 'dod'}
```
[the remark this answers is not in the document]
Comment (crutchfield) [reply PowerShell/ScubaGear/Testing/Unit/PowerShell/Connection/Connect-Tenant.Tests.ps1:8]: Oversight.  Commented out for quicker testing.  Uncomments.

**Review by rgbrow1949 — APPROVED** [review]

**Review by isab-m — APPROVED** [review]
Tested code and validated correct results, everything looks to be running correctly.

**Comment by crutchfield** [comment]
@nanda-katikaneni  PR is ready to merge with main

**Pull request merged**